CI bug: the current rocksdb can't be compile with archlinux:base

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.


### Version

OS: Arch Linux x86_64
Kernel: 5.14.0-162.6.1.el9_1.0.1.x86_64
Kvrocks: 009a1bd80bdde88c091a69aca1539be0f58d2658

### Minimal reproduce step

The CI compiling failed in archlinux:base ("archlinux@sha256:fcc0e315ba72175e77f5c351f1a79ff05fe5548431e842924b7b7cddb8240c08") on rocksdb.
Here is the error message.

```
In file included from /data/kvrocks/build/_deps/rocksdb-src/db/blob/blob_file_meta.cc:6:
/data/kvrocks/build/_deps/rocksdb-src/db/blob/blob_file_meta.h:28:7: error: ‘uint64_t’ has not been declared
   28 |       uint64_t blob_file_number, uint64_t total_blob_count,
      |       ^~~~~~~~
/data/kvrocks/build/_deps/rocksdb-src/db/blob/blob_file_meta.h:15:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   14 | #include "rocksdb/rocksdb_namespace.h"
  +++ |+#include <cstdint>
   15 |
/data/kvrocks/build/_deps/rocksdb-src/db/blob/blob_file_meta.h:28:34: error: ‘uint64_t’ has not been declared
   28 |       uint64_t blob_file_number, uint64_t total_blob_count,
      |                                  ^~~~~~~~
/data/kvrocks/build/_deps/rocksdb-src/db/blob/blob_file_meta.h:28:34: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
/data/kvrocks/build/_deps/rocksdb-src/db/blob/blob_file_meta.h:29:7: error: ‘uint64_t’ has not been declared
   29 |       uint64_t total_blob_bytes, std::string checksum_method,
      |       ^~~~~~~~
/data/kvrocks/build/_deps/rocksdb-src/db/blob/blob_file_meta.h:29:7: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
```

It should fail since the last commit 009a1bd80bdde88c091a69aca1539be0f58d2658.
And this commit could not re-pass the archlinux pipeline now.

### What did you expect to see?

CI passed successfully.

### What did you see instead?

CI failed.
https://github.com/apache/kvrocks/actions/runs/14821772581?pr=2926

### Anything Else?

Maybe we'd better pin an archlinux toolchain version in CI pipeline for stability.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/.github/workflows/kvrocks.yaml
+++ b/.github/workflows/kvrocks.yaml
@@ -485,6 +485,9 @@ jobs:
           pacman -Sy --noconfirm autoconf automake python3 python-redis git wget which cmake make gcc
           echo "NPROC=$(nproc)" >> $GITHUB_ENV
 
+          # FIXME: Workaround for gcc 15 issue for rocksdb https://github.com/apache/kvrocks/issues/2927
+          echo "CXXFLAGS=$CXXFLAGS -include cstdint" >> $GITHUB_ENV
+
       - name: Setup openSUSE
         if: ${{ startsWith(matrix.image, 'opensuse') }}
         run: |

